app/views/application/_navigation_edit_my_application.html.haml

```diff
@@ -1,5 +1,5 @@
 %li.menu-item
-  - if  current_user.is_member? || membership_app.rejected?
-    = link_to t('menus.nav.users.my_application'), membership_application_path(membership_app)
-  - else
+  - if MembershipApplicationPolicy.new(current_user, membership_app).update?
```

Comment:
You have access to the Pundit helper methods in the views (just like in the controllers).
So you can replace this line with:
`if policy(membership_app).update? `

Reply:
That works in the code, but rspec tests will fail with an "undefined method 'policy'" error.

See (e.g.) https://github.com/elabs/pundit/issues/339

Reply:
Hm.  
I was able to get the rspec tests for  to work by putting this into the `before(:)` blocks in `_navigation.html.haml_spec.rb`:

```
describe 'member' do  

  before(:each) do
    ...
    #  policy(membership_app).update?  must be stubbed/mocked @url https://github.com/elabs/pundit/issues/339
    without_partial_double_verification do
      allow(view).to receive(:policy).and_return(double('MembershipApplicationPolicy.new', update?: true))
    end
    ...
  end
...
```
and the same in the `describe 'user with application' do...` before(:each) block


But I didn't do more work to change the tests to ensure that the link for the application is or is not the link to edit the membership application.  

It may be that becomes too hard or not really doable with this problem.  In which case, I'd suggest you just put a comment line (or 2) in the view explaining why you can't use `if policy(membership_app).update?`

Reply:
I took a shortcut by using `MembershipApplicationPolicy.new(current_user, membership_app).update?`. It didn't seem a big deal at the time. But now you've forced me to think about it, I realise I let testing concerns creep into the code - and I shouldn't have.

Thanks for keeping me on the straight and narrow :)